dynamo-lib/src/cmr/dynamo/connection.clj

```diff
@@ -1,35 +1,89 @@
 (ns cmr.dynamo.connection
   "Contains functions for interacting with the DynamoDB storage instance"
-  (:require [cmr.common.config :as cfg :refer [defconfig]]
+  (:require [clojure.string :as str]
+            [cmr.common.config :as cfg :refer [defconfig]]
+            [cmr.common.services.errors :as errors]
+            [cmr.common.log :refer [debug error info trace warn]]
             [taoensso.faraday :as far]
             [cmr.dynamo.config :as dynamo-config]))
 
+(defn concept-revision-tuple->key
+  [tuple]
+  {:concept-id (first tuple) :revision-id (second tuple)})
+
+(defn provider-concept-revision-tuple->key
+  [tuple]
+  {:concept-id (nth tuple 1) :revision-id (nth tuple 2)})
+
+(defn find-params->dynamo-clause
+  ([params]
+   (find-params->dynamo-clause params false))
+  ([params or?]
+   ;; Validate parameter names as a sanity check to prevent sql injection
+   (let [valid-param-name #"^[a-zA-Z][a-zA-Z0-9_\-]*$"]
+     (when-let [invalid-names (seq (filter #(not (re-matches valid-param-name (name %))) (keys params)))]
+       (errors/internal-error! (format "Attempting to search with invalid parameter names [%s]"
+                                       (str/join ", " invalid-names)))))
+   (let [comparisons (for [[k v] params]
+                       (cond
+                         (sequential? v) (let [val (seq v)]
+                                           `(~k [:in ~val]))
+                         (map? v) (let [{:keys [value comparator]} v]
+                                    `(~k [(map-comparator ~comparator) ~value]))
+                         :else `(= ~k ~v)))]
+     (if (> (count comparisons) 1)
+       (if or?
+         (cons `or comparisons)
+         (cons `and comparisons))
+       (first comparisons)))))
+
 (def connection-options 
   {:endpoint (dynamo-config/dynamo-url)})
 
 (defn save-concept
-  [concept])
+  [concept]
+  (info "Concept passed to dynamo/save-concept: " concept)
+  (far/put-item connection-options (dynamo-config/dynamo-table) concept))
 
 (defn get-concept
   "Gets a concept from DynamoDB"
-  ([provider concept-type concept-id]))
+  ([concept-id]
+   (first (far/query connection-options (dynamo-config/dynamo-table) {:concept-id [:eq concept-id]} {:order :desc :limit 1})))
+  ([concept-id revision-id]
+   (far/get-item connection-options (dynamo-config/dynamo-table) {:concept-id concept-id :revision-id revision-id})))
 
-(defn get-concepts
+(defn get-concepts-provided
   "Gets a group of concepts from DynamoDB"
-  [provider concept-type concept-id-revision-id-tuples]
-  ())
+  [concept-id-revision-id-tuples]
+  (remove nil? (doall (map (fn [batch] (far/batch-get-item connection-options {(dynamo-config/dynamo-table) {:prim-kvs (vec batch)}})) (partition-all 100 (map concept-revision-tuple->key concept-id-revision-id-tuples))))))
```

Comment:
can any of the "remove-nil..." functions from cmr.common.util be used here instead of rolling your own?

Reply:
Was not aware there were any, I'll take a look into it

Reply:
Also, why 100? Why not 2000? Is 100 special?

Reply:
Yep, that is a limit set by DynamoDB. Batch-get-item is limited to batches of 100, batch-write-item is limited to batches of size 25